subprojects/core-api/src/main/java/org/gradle/api/component/SoftwareComponent.java

```diff
@@ -21,14 +21,9 @@
 
 /**
  * A software component produced by a Gradle software project.
- *
- * <p>An implementation of this interface may also implement:</p>
- *
- * <ul>
- *
- * <li>{@link ComponentWithVariants} to provide information about the variants that the component provides.</li>
- *
- * </ul>
+ * <p>
+ * An implementation of this interface may also implement {@link ComponentWithVariants} to provide
```

Comment:
I think this might be too deep in the weeds for this comment.  They are instances of `SoftwareComponent` yes, but they are referred to in the method and other javadoc as variants, and conceptually, this is currently how you add variants.

I think making this distinction is better suited for the docs on `ComponentWithVariants` itself, which does already mention both that they are children and variants.

This change is just fixing the formatting to get rid of an unnecessary 1 item list, the text is the same.  I'm going to leave this as is, but if you want to send me some better comment text I can add that to this PR.